plugins/woocommerce/tests/e2e-pw/tests/merchant/products/block-editor/linked-product-tab-product-block-editor.spec.js

```diff
@@ -0,0 +1,165 @@
+const { test } = require( '../../../../fixtures/block-editor-fixtures' );
+const { expect } = require( '@playwright/test' );
+
+const { clickOnTab } = require( '../../../../utils/simple-products' );
+const { api } = require( '../../../../utils' );
+
+const NEW_EDITOR_ADD_PRODUCT_URL =
+	'wp-admin/admin.php?page=wc-admin&path=%2Fadd-product';
+
+const isTrackingSupposedToBeEnabled = !! process.env.ENABLE_TRACKING;
+
+const productData = {
+	name: `Linked product Name ${ new Date().getTime().toString() }`,
+	summary: 'This is a product summary',
+};
+
+const linkedProductsData = [],
+	productIds = [];
+let productId = 0;
+
+test.describe( 'General tab', () => {
+	test.describe( 'Linked product', () => {
+		test.beforeAll( async () => {
+			for ( let i = 1; i <= 5; i++ ) {
+				const product = {
+					name: `Product name ${ i } ${ new Date()
+						.getTime()
+						.toString() }`,
+					productPrice: `${ i }00`,
+					type: 'simple',
+				};
+				linkedProductsData.push( product );
+				const id = await api.create.product( product );
+				productIds.push( id );
+			}
+		} );
+
+		test.afterAll( async () => {
+			for ( const aProductId of productIds ) {
+				await api.deletePost.product( aProductId );
+			}
+			await api.deletePost.product( productId );
+		} );
+		test.skip(
+			isTrackingSupposedToBeEnabled,
+			'The block product editor is not being tested'
+		);
+
+		test( 'can create a product with linked products', async ( {
+			page,
+		} ) => {
+			await page.goto( NEW_EDITOR_ADD_PRODUCT_URL );
+			await clickOnTab( 'General', page );
+			await page
+				.getByPlaceholder( 'e.g. 12 oz Coffee Mug' )
+				.fill( productData.name );
+			await page
+				.locator(
+					'[data-template-block-id="basic-details"] .components-summary-control'
+				)
+				.last()
+				.fill( productData.summary );
+
+			await clickOnTab( 'Linked products', page );
+
+			await page
+				.getByRole( 'heading', {
+					name: 'Cross-sells',
+				} )
+				.isVisible();
+
+			await page
+				.locator(
+					'.wp-block-woocommerce-product-linked-list-field__form-group-content'
+				)
+				.first()
+				.getByPlaceholder( 'Search for products' )
+				.fill( linkedProductsData[ 0 ].name );
+
+			await page.getByText( linkedProductsData[ 0 ].name ).click();
+
+			await page.getByText( 'Choose products for me' ).first().click();
+
+			await page.waitForResponse(
+				( response ) =>
+					response
+						.url()
+						.includes(
+							'/wp-json/wc/v3/products?search=&orderby=title&order=asc&per_page='
+						) && response.status() === 200
+			);
```

Comment:
The `waitForResponse` promise should be initiated before the action that triggers the request is performed. Otherwise, due to the async nature of the framework, we might end up with a response that already happened before we started waiting for it, hence making the test flaky. The correct sequence in situations like this would be:

```js
		const chooseProductsResponsePromise = page.waitForResponse(
			( response ) =>
				response
					.url()
					.includes(
						'/wp-json/wc/v3/products?search=&orderby=title&order=asc&per_page='
					) && response.status() === 200
		);

		await page.getByText( 'Choose products for me' ).first().click();
		await chooseProductsResponsePromise;
```

This way, we ensure the request/response is caught as we start waiting before the triggering action is performed.

See https://playwright.dev/docs/api/class-page#page-wait-for-response for more info!